packages/amplify_authenticator/lib/amplify_authenticator.dart

```diff
@@ -202,28 +207,54 @@ class Authenticator extends StatefulWidget {
 
 class _AuthenticatorState extends State<Authenticator> {
   final AuthService _authService = AmplifyAuthService();
-
   late final StateMachineBloc _stateMachineBloc;
+  AmplifyConfig? _config;
+  late List<String> _missingConfigValues;
+  bool _configInitialized = false;
 
   @override
   void initState() {
     super.initState();
     _stateMachineBloc = StateMachineBloc(_authService)
       ..authEvent.add(GetCurrentUser());
+    waitForConfiguration();
+  }
+
+  Future<void> waitForConfiguration() async {
+    await Amplify.asyncConfig.then((config) {
+      setState(() {
+        _config = config;
+        _configInitialized = true;
+        _missingConfigValues = missingConfigValues(config);
+      });
+    });
+  }
+
+  List<String> missingConfigValues(AmplifyConfig? config) {
+    List<String> missingValues = [];
+    var cognitoPlugin = config?.auth?.awsCognitoAuthPlugin?.auth?['Default'];
+    cognitoPlugin ?? missingValues.add('auth.plugins.Auth.Default');
+    cognitoPlugin?.signupAttributes?.length ??
+        missingValues.add('auth.plugins.Auth.Default.signUpAttributes');
+    return missingValues;
   }
 
   @override
   Widget build(BuildContext context) {
     var defaultForms = DefaultForms();
     AuthStringResolver resolver = widget.resolver ?? AuthStringResolver();
-
     defaultForms.context = context;
 
+    if (_configInitialized && _missingConfigValues.isNotEmpty) {
+      throw StateError(
+          'Encountered problem(s) building the Authenticator because your amplifyconfiguration.dart file is missing required values: ${_missingConfigValues.join('/n')}). You should correct your amplifyconfiguration.dart file and restart your app.');
```

Comment:
Prefer a multiline string here

Reply:
Also, I feel like this might be TMI for the typical authenticator user. What about just saying "please re-run `amplify pull`"?

Reply:
+1 on including some info on how to correct it. I do think that including the missing config values could be helpful in some scenarios.

Maybe something like 

> Encountered problem(s) building the Authenticator because your amplifyconfiguration.dart file is missing required values. Please re-run `amplify pull` and restart your app.
> 
> Missing values are: values ...

Reply:
I think we should keep CLI references out of the code, since customers are not required to use the CLI at all in order to use amplify-flutter.